amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala

```diff
@@ -13,7 +14,7 @@ trait JsonSchemaInference {
     parseSchemaEntry(ast) match {
       case Some(node) =>
         tryParserVersion(node)(errorHandler) match {
-          case Some(version) => getVersionFor(version).getOrElse(defaultSchemaVersion)
+          case Some(version) => JsonSchemaEntry(version).getOrElse(defaultSchemaVersion)
```

Comment:
The parent trait is calling the object that implement it. This is weird, I don't know why this is compiling...

Reply:
done, moved logic to parent